keepassxc-browser/keepassxc-browser.js

```diff
@@ -1126,7 +1126,34 @@ cipFields.useDefinedCredentialFields = function() {
     return false;
 };
 
+MutationObserver = window.MutationObserver || window.WebKitMutationObserver;
+
+// Detects DOM changes in the document
+let observer = new MutationObserver(function(mutations, observer) {
+    if (document.visibilityState !== 'hidden') {
+        for (const mut of mutations) {
+            if (mut.type === 'childList') {
+                const fields = cipFields.getAllFields();
+
+                // If only password field is shown it's enough to have one field visible for initCredentialFields
+                if (fields.length > (_detectedFields == 1 ? 0 : 1)) {
+                    cip.initCredentialFields(true);
+                }
+            }
+        }
+    }
+});
 
+// define what element should be observed by the observer
+// and what types of mutations trigger the callback
+observer.observe(document, {
+    subtree: true,
+    attributes: true,
+    childList: true,
+    characterData: true,
+    attributeOldValue: true,
+    characterDataOldValue: true

```

Comment:
This list contains unnecessary items. I don't believe listening for characterdata, attributeoldvalue, and characteroldvalue would ever be involved in the appearance of new form fields. Also, is it possible to only listen for mutations on existing (and new) form blocks? It seems very wasteful to listen to the entire document.

Reply:
True, I removed a few of those. The [documentation](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#MutationObserverInit) says:
```
Note: At the very least, childList, attributes, or characterData must be set to true. Otherwise, "An invalid or illegal string was specified" error is thrown.
```

Reply:
Also, listening the entire document is necessary because otherwise new div's added to it wouldn't be detected at all.